encoding/encode_test.go

```diff
@@ -16,9 +16,91 @@
 
 package encoding_test
 
-import "testing"
+import (
+	"bytes"
+	"testing"
+
+	"errors"
+	"fmt"
+
+	"github.com/DE-labtory/koa/encoding"
+)
 
 // TODO: implement test cases :-)
 func TestEncodeOperand(t *testing.T) {
+	tests := []struct {
+		operand      interface{}
+		expectedByte []byte
+		expectedErr  error
+	}{
+		{
+			operand:      true,
+			expectedByte: []byte{0x00, 0x00, 0x00, 0x01},
+			expectedErr:  nil,
+		},
+		{
+			operand:      false,
+			expectedByte: []byte{0x00, 0x00, 0x00, 0x00},
+			expectedErr:  nil,
+		},
+		{
+			operand:      "abc",
+			expectedByte: []byte{0x61, 0x62, 0x63},
+			expectedErr:  nil,
+		},
+		{
+			operand:      "1234567890",
+			expectedByte: []byte{0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x30},
+			expectedErr:  nil,
+		},
+		{
+			operand:      "~!@#$%^&*()_+",
+			expectedByte: []byte{0x7e, 0x21, 0x40, 0x23, 0x24, 0x25, 0x5e, 0x26, 0x2a, 0x28, 0x29, 0x5f, 0x2b},
+			expectedErr:  nil,
+		},
+		{
+			operand:      "12!@qw",
+			expectedByte: []byte{0x31, 0x32, 0x21, 0x40, 0x71, 0x77},
+			expectedErr:  nil,
+		},
+		{
+			operand:      1,
+			expectedByte: []byte{0x00, 0x00, 0x00, 0x01},
+			expectedErr:  nil,
+		},
+		{
+			operand:      23,
+			expectedByte: []byte{0x00, 0x00, 0x00, 0x17},
+			expectedErr:  nil,
+		},
+		{
+			operand:      456,
+			expectedByte: []byte{0x00, 0x00, 0x01, 0xc8},
+			expectedErr:  nil,
+		},
+		{
+			operand:      'c',
+			expectedByte: nil,
+			expectedErr:  errors.New(fmt.Sprintf("EncodeOperand() error - operand %v could not encoded", 'c')),
+		},
+	}
+
+	for i, test := range tests {
+		b, err := encoding.EncodeOperand(test.operand)
+
+		if !bytes.Equal(b, test.expectedByte) {
+			t.Fatalf("test[%d] - EncodeOperand() result wrong. expected=%x, got=%x", i, test.expectedByte, b)
+		}
 
+		switch err.(type) {
+		case nil:
```

Comment:
Why did you separate error type? What is difference between `type nil` error and `type error` error??